.github/workflows/ci.yml

```diff
@@ -0,0 +1,113 @@
+name: CI
+on: [push, pull_request]
+env:
+  HUGO_VERSION: 0.26
+  SCALA_VERSION: 2.12.10
+jobs:
+  validate:
+    name: Scala ${{ matrix.scala }}, Java ${{ matrix.java }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [8, 11]
+        scala: [2.12.10, 2.13.1]
+    env:
+      SCALA_VERSION: ${{ matrix.scala }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache Coursier
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/coursier
+          key: sbt-coursier-cache
+      - name: Cache SBT
+        uses: actions/cache@v1
+        with:
+          path: ~/.sbt
+          key: sbt-${{ hashFiles('**/build.sbt') }}
+      - name: Check formatting
+        if: startsWith(matrix.scala, '2.12.') && (matrix.java == '8')
+        run: sbt ++$SCALA_VERSION scalafmtCheckAll
+      - name: Compile
+        run: sbt ++$SCALA_VERSION test:compile
+      - name: Check compatibility
+        run: sbt ++$SCALA_VERSION mimaReportBinaryIssues
+      - name: Check unused dependencies
+        run: sbt ++$SCALA_VERSION unusedCompileDependenciesTest
+      - name: Test
+        run: sbt ++$SCALA_VERSION test
+      - name: Scaladoc
+        run: sbt ++$SCALA_VERSION doc
```

Comment:
Splitting all these steps makes it clearer where it fails, but costs us ~20 seconds to start a new SBT each time.

Reply:
You mean being able to see the failure stage in the GitHub UI, as opposed to looking in the console output to see where it failed?

Reply:
Yes,  exactly. I guess we could time it with and without and see what it costs.